### ISSUE TITLE
Consumer does not keep listening using - spring-cloud-gcp-pubsub-stream-binder WITHOUT web support

### ISSUE THREAD

**Issue by rapar8** [GoogleCloudPlatform/spring-cloud-gcp#1182]
keep Spring boot pubsub stream application running without Web server (tomcat).

Application works great with Tomcat(spring-boot-starter-web) but If I remove web support it is not keep listening to but shutdown.

Spring boot 2.7.1

spring-cloud-gcp-pubsub-stream-binder 3.3.0

All details are mentioned in [Stackoverflow](https://stackoverflow.com/questions/72868966/spring-stream-with-gcp-consumer-only-keep-project-running-without-web-app-only)
[pubsub-publisher.zip](https://github.com/GoogleCloudPlatform/spring-cloud-gcp/files/9070037/pubsub-publisher.zip)
[pubsub-consumer.zip](https://github.com/GoogleCloudPlatform/spring-cloud-gcp/files/9070043/pubsub-consumer.zip)

**Comment by elefeint** [comment]
This behavior is intentional, although I agree that it's unintuitive in the context of a Spring Cloud Stream application. There is a note about this [in the docs](https://googlecloudplatform.github.io/spring-cloud-gcp/3.3.0/reference/html/index.html#subscribe-methods) -- 

> As of version 1.2, subscribing by itself is not enough to keep an application running. For a command-line application, you may want to provide your own ThreadPoolTaskScheduler bean named pubsubSubscriberThreadPool, which by default creates non-daemon threads that will keep an application from stopping. This default behavior has been overridden in Spring Cloud GCP for consistency with Cloud Pub/Sub client library, and to avoid holding up command-line applications that would like to shut down once their work is done

**Comment by elefeint** [comment]
Reopening this because some bean names changed recently, and the documentation might not be up to date anymore.

**Comment by elefeint** [comment]
For your immediate need, adding these two beans should keep the application alive:

```
	@Bean
	public ThreadPoolTaskScheduler pubsubSubscriberThreadPool() {
		ThreadPoolTaskScheduler scheduler = new ThreadPoolTaskScheduler();
		scheduler.setPoolSize(4);  // customize as needed
		scheduler.setThreadNamePrefix("custom-pubsub-subscriber");
		return scheduler;
	}

	@Bean
	public ExecutorProvider subscriberExecutorProvider(
			@Qualifier("pubsubSubscriberThreadPool") ThreadPoolTaskScheduler scheduler) {
		return FixedExecutorProvider.create(scheduler.getScheduledExecutor());
	}
```

I am going to keep this issue open because we need to update the documentation, and consider either not deprecating this global property or (more likely) introducing a property for keeping autoconfiguration-created thread pools alive.

@mpeddada1 FYI.

**Comment by rapar8** [comment]
I have added this 2 beans but it is still not keeping app Up.

`2022-07-12 11:50:52.407  INFO 4019 --- [ionShutdownHook] .g.c.s.p.i.i.PubSubInboundChannelAdapter : stopped com.google.cloud.spring.pubsub.integration.inbound.PubSubInboundChannelAdapter@670342a2
2022-07-12 11:50:52.408  INFO 4019 --- [ionShutdownHook] o.s.i.monitor.IntegrationMBeanExporter   : Registering MessageChannel my-topic.myGroup1.errors
2022-07-12 11:50:52.438  INFO 4019 --- [ionShutdownHook] o.s.c.stream.binder.BinderErrorChannel   : Channel 'application.my-topic.myGroup1.errors' has 1 subscriber(s).
2022-07-12 11:50:52.439  INFO 4019 --- [ionShutdownHook] o.s.i.endpoint.EventDrivenConsumer       : Removing {logging-channel-adapter:_org.springframework.integration.errorLogger} as a subscriber to the 'errorChannel' channel
2022-07-12 11:50:52.439  INFO 4019 --- [ionShutdownHook] o.s.i.channel.PublishSubscribeChannel    : Channel 'application.errorChannel' has 0 subscriber(s).
2022-07-12 11:50:52.440  INFO 4019 --- [ionShutdownHook] o.s.i.endpoint.EventDrivenConsumer       : stopped bean '_org.springframework.integration.errorLogger'
`

**Comment by elefeint** [comment]
As a workaround, add `@EnableScheduling` to your application, and add a bean like this to keep the application running:
```
  @Scheduled (fixedRate = 1, timeUnit = TimeUnit.MINUTES)
  public void ensureApplicationRuns() {
    // do nothing, or log message
  }
```

**Comment by meltsufin** [comment]
Fixed in #1404.
